.github/workflows/macos.yaml

```diff
@@ -112,16 +112,21 @@ jobs:
     - name: Clone datadog-agent repository
       env:
         VERSION: ${{ github.event.inputs.datadog_agent_ref || env.DEFAULT_DATADOG_AGENT_REF }}
-      run: |
-        bash ./scripts/clone_agent.sh
-        cd $HOME/go/src/github.com/DataDog/datadog-agent
-        export VERSION_CACHE_CONTENT="${{github.event.inputs.version_cache}}"
-        if [ ! -z "$VERSION_CACHE_CONTENT" ]; then base64 -d <(echo "$VERSION_CACHE_CONTENT") > ./agent-version.cache; fi
-        # Copy files used for the Omnibus ruby deps cache key
-        # This is to work around a limitation of GitHub Actions' `hashFiles()` function which requires
-        # files to be inside the workspace.
-        mkdir ${GITHUB_WORKSPACE}/_omnibus_cache_key_files
-        cp $HOME/go/src/github.com/DataDog/datadog-agent/{release.json,omnibus/Gemfile} ${GITHUB_WORKSPACE}/_omnibus_cache_key_files
+      uses: nick-fields/retry@v3
```

Comment:
```suggestion
      uses: nick-fields/retry@7152eba30c6575329ac0576536151aca5a72780e # v3.0.0
```
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Same as above